airbyte-integrations/connectors/source-okta/unit_tests/test_source.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from unittest.mock import MagicMock
+
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from source_okta.source import (
+    CustomRoles,
+    GroupMembers,
+    GroupRoleAssignments,
+    Groups,
+    Logs,
+    OktaOauth2Authenticator,
+    SourceOkta,
+    UserRoleAssignments,
+    Users,
+)
+from unit_tests import (
+    auth_token_config,
+    oauth_config,
+    token_config,
+    wrong_oauth_config_bad_auth_type,
+    wrong_oauth_config_bad_credentials_record,
+)
+
+
+class TestAuthentication:
+    def test_init_token_authentication_init(self):
+        config = token_config()
+        source_okta = SourceOkta()
+        token_authenticator_instance = source_okta.initialize_authenticator(config=config)
+        assert isinstance(token_authenticator_instance, TokenAuthenticator)
+
+        config = auth_token_config()
+        token_authenticator_instance = source_okta.initialize_authenticator(config=config)
+        assert isinstance(token_authenticator_instance, TokenAuthenticator)
+
+    def test_init_oauth2_authentication_init(self):
+        config = oauth_config()
+        source_okta = SourceOkta()
+        oauth_authentication_instance = source_okta.initialize_authenticator(config=config)
+        assert isinstance(oauth_authentication_instance, OktaOauth2Authenticator)
+
+    def test_init_oauth2_authentication_wrong_credentials_record(self):
+        config = wrong_oauth_config_bad_credentials_record()
+        source_okta = SourceOkta()
+        try:
+            source_okta.initialize_authenticator(config=config)
+        except Exception as e:
+            assert e.args[0] == "Config validation error. `credentials` not specified."
+
+    def test_init_oauth2_authentication_wrong_oauth_config_bad_auth_type(self):
+        config = wrong_oauth_config_bad_auth_type()
+        source_okta = SourceOkta()
+        try:
+            source_okta.initialize_authenticator(config=config)
+        except Exception as e:
+            assert e.args[0] == "Config validation error. `auth_type` not specified."
+
+    def test_check_connection_ok(self, requests_mock):
+        config = oauth_config()
```

Comment:
Try to use fixture to correspond DRY principle

Reply:
done